criu/files-reg.c

```diff
@@ -1301,9 +1267,11 @@ static int check_path_remap(struct fd_link *link, const struct fd_parms *parms,
 		 * name.
 		 */
 
-		if (errno == ENOENT)
+		if (errno == ENOENT) {
+			link_strip_deleted(link);
```

Comment:
Probably you should port https://src.openvz.org/projects/OVZ/repos/criu/commits/f5768e79fd ("files: Don't forget on stipping deleted postfix on linked files") as a separate patch instead of implicitly adding link_strip_deleted here, for the sake of better git history.